server/prisma/schema.prisma

```diff
@@ -907,3 +909,38 @@ model Config {
   value     String   @db.Text
   updatedAt DateTime @default(now()) @db.Timestamp()
 }
+
+// CLAN ROLES
+model ClanRole {
+  id                       String           @id @unique @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
+  clan                     Clan             @relation(fields: [clanId], references: [id], onDelete: Cascade)
+  clanId                   String           @db.Uuid
+  name                     String           @db.VarChar(50)
+  // Permissions
+  canAcceptJoinRequests    Boolean          @default(false)
+  canRejectJoinRequests    Boolean          @default(false)
+  canRemoveMembers         Boolean          @default(false)
+  canSelectWarFighters     Boolean          @default(false)
+  canPinThreads            Boolean          @default(false)
+  canUnpinThreads          Boolean          @default(false)
+  canDeletePosts           Boolean          @default(false)
+  canDeleteThreads         Boolean          @default(false)
+  canCreateRoles           Boolean          @default(false)
+  canChangeRoles           Boolean          @default(false)
+  members                  ClanMemberRole[]
+  createdAt                DateTime         @default(now()) @db.Timestamp()
+
+  @@index([clanId])
```

Comment:
I haven't checked the queries you use in the server yet, but I'm pretty sure you're gonna need more indexes (composites too) than that